src/quat.js

```diff
@@ -447,32 +447,75 @@ export function fromMat3(out, m) {
 }
 
 /**
- * Creates a quaternion from the given euler angle x, y, z.
+ * Creates a quaternion from the given euler angle x, y, z using the provided extrinsic order for the conversion.
  *
  * @param {quat} out the receiving quaternion
  * @param {x} Angle to rotate around X axis in degrees.
- * @param {y} Angle to rotate around Y axis in degrees.
  * @param {z} Angle to rotate around Z axis in degrees.
+ * @param {y} Angle to rotate around Y axis in degrees.
+ * @param {order} Intrinsic order for conversion, so far only 'xyz' and 'zyx' is supported.
  * @returns {quat} out
  * @function
  */
-export function fromEuler(out, x, y, z) {
+export function fromEuler(out, x, y, z, order='zyx') {
   let halfToRad = (0.5 * Math.PI) / 180.0;
   x *= halfToRad;
-  y *= halfToRad;
   z *= halfToRad;
+  y *= halfToRad;
 
   let sx = Math.sin(x);
   let cx = Math.cos(x);
-  let sy = Math.sin(y);
-  let cy = Math.cos(y);
   let sz = Math.sin(z);
   let cz = Math.cos(z);
+  let sy = Math.sin(y);
+  let cy = Math.cos(y);
 
-  out[0] = sx * cy * cz - cx * sy * sz;
-  out[1] = cx * sy * cz + sx * cy * sz;
-  out[2] = cx * cy * sz - sx * sy * cz;
-  out[3] = cx * cy * cz + sx * sy * sz;
+  if (typeof order != 'string') {
+    order = 'zyx';
+  }
```

Comment:
nit: because you're using `default` in the switch case this condition isn't needed.

Reply:
It's so `toLowerCase()` will not throw a TypeError.